src/main/java/org/folio/service/pieces/flows/update/PieceUpdateFlowInventoryManager.java

```diff
@@ -45,73 +47,62 @@ public PieceUpdateFlowInventoryManager(TitlesService titlesService,
-      .map(itemId -> {
-        Optional.ofNullable(itemId).ifPresent(holder.getPieceToUpdate()::withItemId);
-        return null;
-      })
-      .compose(aVoid -> deleteHolding(holder, requestContext))
-      .onSuccess(pair -> logger.debug(UPDATE_INVENTORY_FOR_LINE_DONE))
-      .mapEmpty();
+  private Future<String> getPoLineInstanceId(CompositePoLine compPOL) {
+    return compPOL.getInstanceId() != null || PoLineCommonUtil.isInventoryUpdateNotRequired(compPOL)
+      ? Future.succeededFuture(compPOL.getInstanceId())
+      : null;
   }
 
   private Future<Pair<String, String>> deleteHolding(PieceUpdateHolder holder, RequestContext requestContext) {
-    if (holder.isDeleteHolding()) {
-      return pieceUpdateInventoryService.deleteHoldingConnectedToPiece(holder.getPieceFromStorage(), requestContext);
-    }
-    return Future.succeededFuture();
+    return holder.isDeleteHolding()
+      ? pieceUpdateInventoryService.deleteHoldingConnectedToPiece(holder.getPieceFromStorage(), requestContext)
+      : Future.succeededFuture();
   }
 
   private Future<Location> handleHolding(PieceUpdateHolder holder, RequestContext requestContext) {
-    CompositePoLine poLineToSave = holder.getPoLineToSave();
-    Piece pieceToUpdate = holder.getPieceToUpdate();
+    var pieceToUpdate = holder.getPieceToUpdate();
     if (pieceToUpdate.getHoldingId() != null) {
       return Future.succeededFuture(new Location().withHoldingId(pieceToUpdate.getHoldingId()));
     }
-    String instanceId = holder.getInstanceId();
-    if (instanceId != null && DefaultPieceFlowsValidator.isCreateHoldingForPiecePossible(pieceToUpdate, poLineToSave)) {
-      return inventoryHoldingManager.createHoldingAndReturnId(instanceId, pieceToUpdate.getLocationId(), requestContext)
-        .map(holdingId -> {
-          Location location = new Location().withLocationId(pieceToUpdate.getLocationId());
-          Optional.ofNullable(holdingId).ifPresent(holdingIdP -> {
-            pieceToUpdate.setLocationId(null);
-            pieceToUpdate.setHoldingId(holdingId);
-            location.setLocationId(null);
-            location.setHoldingId(holdingId);
-          });
-          return location;
-        });
+
+    var poLineToSave = holder.getPoLineToSave();
+    var instanceId = holder.getInstanceId();
+    var location = new Location().withLocationId(pieceToUpdate.getLocationId());
+    if (instanceId == null || !DefaultPieceFlowsValidator.isCreateHoldingForPiecePossible(pieceToUpdate, poLineToSave)) {
+      return Future.succeededFuture(location);
     }
-    return Future.succeededFuture(new Location().withLocationId(pieceToUpdate.getLocationId()));
+    return inventoryHoldingManager.createHoldingAndReturnId(instanceId, pieceToUpdate.getLocationId(), requestContext)
+      .map(holdingId -> {
+        if (holdingId != null) {
+          pieceToUpdate.withHoldingId(null).setHoldingId(holdingId);
+          location.withLocationId(null).setHoldingId(holdingId);
```

Comment:
is it supposed to be like this `pieceToUpdate.withLocationId(null).setHoldingId(holdingId)`

Reply:
That's right, thanks for pointing out